psdk_wrapper/src/psdk_wrapper.cpp

```diff
@@ -87,30 +88,57 @@ PSDKWrapper::PSDKWrapper(const std::string &node_name)
   declare_parameter("data_frequency.esc_data_frequency", 1);
   declare_parameter("num_of_initialization_retries", 1);
 
-  // Create module nodes
-  flight_control_module_ =
-      std::make_shared<FlightControlModule>("flight_control_node");
-  telemetry_module_ = std::make_shared<TelemetryModule>("telemetry_node");
-  psdk_ros2::global_telemetry_ptr_ = telemetry_module_;
-  camera_module_ = std::make_shared<CameraModule>("camera_node");
-  psdk_ros2::global_camera_ptr_ = camera_module_;
-  liveview_module_ = std::make_shared<LiveviewModule>("liveview_node");
-  psdk_ros2::global_liveview_ptr_ = liveview_module_;
-  gimbal_module_ = std::make_shared<GimbalModule>("gimbal_node");
-  hms_module_ = std::make_shared<HmsModule>("hms_node");
-  psdk_ros2::global_hms_ptr_ = hms_module_;
-  perception_module_ = std::make_shared<PerceptionModule>("perception_node");
-  psdk_ros2::global_perception_ptr_ = perception_module_;
-
-  // Start the threads
-  telemetry_thread_ = std::make_unique<utils::NodeThread>(telemetry_module_);
-  flight_control_thread_ =
-      std::make_unique<utils::NodeThread>(flight_control_module_);
-  camera_thread_ = std::make_unique<utils::NodeThread>(camera_module_);
-  liveview_thread_ = std::make_unique<utils::NodeThread>(liveview_module_);
-  gimbal_thread_ = std::make_unique<utils::NodeThread>(gimbal_module_);
-  hms_thread_ = std::make_unique<utils::NodeThread>(hms_module_);
-  perception_thread_ = std::make_unique<utils::NodeThread>(perception_module_);
+  load_parameters();
```

Comment:
isn't the load parameters an issue  as inside this function there are some parameters that try to access e.g. telemetry module objects which have not been created yet. I think you could only retrieve the is module mandatory parameters here and leave the load_parameters() in the same place where it was in on_configure